fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/rules/ImplementTypeFilterRule.java

```diff
@@ -82,7 +82,7 @@ public void onMatch(@Nonnull final CascadesRuleCall call) {
         final var unsatisfiedMapBuilder = ImmutableMultimap.<Set<String>, RecordQueryPlan>builder();
 
         for (final var innerPlan : planPartition.getPlans()) {
-            final var childRecordTypes = RecordTypesProperty.evaluate(call.newAliasResolver(), innerPlan);
+            final var childRecordTypes = new RecordTypesVisitor(call.newAliasResolver()).visit(innerPlan);
```

Comment:
Is there a reason this is constructing a new `ReocrdTypesVisitor` directly instead of evaluating the property on the inner plan? Is it just the alias resolver? And is that actually needed for the property evaluation?

Reply:
The property cannot be used as the visitor needs an argument.

Reply:
To be clear, I guess I'm fine with leaving the argument in under the doctrine of making minimal changes

Reply:
Yes -- this is not really a property call. The visitor needs an argument (in this case) in the shape of an alias resolver. That resolver can follow up deep correlations in order to ensure that a datum can only come from a set of record types. Now it's debatable if this is right in this case or this should be handled differently (you can use referential typing but that's another can of worms). To keep the status quo, this visitor needs the alias resolver.